luxonis_train/nodes/base_node.py

```diff
@@ -341,24 +439,28 @@ def run(self, inputs: list[Packet[Tensor]]) -> Packet[Tensor]:
         """
         unwrapped = self.unwrap(self.validate(inputs))
         outputs = self(unwrapped)
-        return self.wrap(outputs)
+        wrapped = self.wrap(outputs)
+        str_tasks = [task.value for task in self._tasks] if self._tasks else []
+        for key in list(wrapped.keys()):
+            if key in str_tasks:
+                value = wrapped.pop(key)
+                wrapped[self.get_task_name(LabelType(key))] = value
+        return wrapped
 
     def validate(self, data: list[Packet[Tensor]]) -> list[Packet[Tensor]]:
-        """Validates the inputs against `in_protocols`."""
-        if len(data) != len(self.in_protocols):
+        """Validates the inputs against `inpur_protocols`."""
```

Comment:
Typo in docstring "inpur"
```suggestion
        """Validates the inputs against `input_protocols`."""
```